subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCleanupService.java

```diff
@@ -22,55 +22,47 @@
 import org.gradle.internal.file.Deleter;
 import org.gradle.internal.logging.progress.ProgressLogger;
 import org.gradle.internal.logging.progress.ProgressLoggerFactory;
-import org.gradle.util.internal.GUtil;
 
 import java.io.File;
-import java.util.Properties;
 
 public class GradleUserHomeCleanupService implements Stoppable {
 
     private static final long MAX_UNUSED_DAYS_FOR_RELEASES = 30;
     private static final long MAX_UNUSED_DAYS_FOR_SNAPSHOTS = 7;
-    private static final String CACHE_CLEANUP_PROPERTY = "org.gradle.cache.cleanup";
 
     private final Deleter deleter;
     private final GradleUserHomeDirProvider userHomeDirProvider;
     private final GlobalScopedCache globalScopedCache;
     private final UsedGradleVersions usedGradleVersions;
     private final ProgressLoggerFactory progressLoggerFactory;
+    private final CacheCleanupEnablement cacheCleanupEnablement;
 
     public GradleUserHomeCleanupService(
         Deleter deleter,
         GradleUserHomeDirProvider userHomeDirProvider,
         GlobalScopedCache globalScopedCache,
         UsedGradleVersions usedGradleVersions,
-        ProgressLoggerFactory progressLoggerFactory
+        ProgressLoggerFactory progressLoggerFactory,
+        CacheCleanupEnablement cacheCleanupEnablement
     ) {
         this.deleter = deleter;
         this.userHomeDirProvider = userHomeDirProvider;
         this.globalScopedCache = globalScopedCache;
         this.usedGradleVersions = usedGradleVersions;
         this.progressLoggerFactory = progressLoggerFactory;
+        this.cacheCleanupEnablement = cacheCleanupEnablement;
     }
 
     @Override
     public void stop() {
-        // TODO Will be implemented without hard-coded access to `$GRADLE_USER_HOME/gradle.properties` for 5.1 in #6084
-        File gradleUserHomeDirectory = userHomeDirProvider.getGradleUserHomeDirectory();
-        File gradleProperties = new File(gradleUserHomeDirectory, "gradle.properties");
-        if (gradleProperties.isFile()) {
-            Properties properties = GUtil.loadProperties(gradleProperties);
-            String cleanup = properties.getProperty(CACHE_CLEANUP_PROPERTY);
-            if (cleanup != null && cleanup.equals("false")) {
-                return;
+        if (cacheCleanupEnablement.isEnabled()) {
```

Comment:
Basically, the only way to affect this is to change the value in GUH/gradle.properties.  So, yes, it could change in between two builds, but it really only matters when these services are stopped.  The way I understand it, these services get reused as long as the same GUH is in use and would only stop when 1) the daemon stops or 2) a different GUH was being used.  It's not immediately clear to me when 2 would actually happen, but in either case, I don't think there's any danger to this value changing in between builds.  So unless I'm misunderstanding the question, I don't think this is an issue.  I'm happy to investigate further if you're thinking of a scenario that I'm not.